internal/pkg/build/sources/oci_unpack_linux.go

```diff
@@ -68,5 +70,41 @@ func unpackRootfs(b *sytypes.Bundle, tmpfsRef types.ImageReference, sysCtx *type
 	os.RemoveAll(b.Rootfs())
 
 	// Unpack root filesystem
-	return umocilayer.UnpackRootfs(context.Background(), engineExt, b.Rootfs(), manifest, &mapOptions)
+	err = umocilayer.UnpackRootfs(context.Background(), engineExt, b.Rootfs(), manifest, &mapOptions)
+	if err != nil {
+		return err
+	}
+
+	// Fix permissions to allow the rootfs to be moved and deleted by the user
+	// if we performed a rootless extraction (#4517)
+	if mapOptions.Rootless {
+		sylog.Debugf("Fixing rootless permissions on %s", b.Rootfs())
+		return fixPermsRootless(b.Rootfs())
+	}
+
+	return nil
+
+}
+
+// fixPermsRootless forces permissions on the rootfs so that it can be easily
+// moved and deleted by a non-root user owner.
+func fixPermsRootless(rootfs string) (err error) {
+	err = filepath.Walk(rootfs, func(path string, f os.FileInfo, err error) error {
+		// Directories must have the owner 'rx' bits to allow traversal and reading on move, and the 'w' bit
+		// so their content can be deleted by the user when the rootfs/sandbox is deleted
+		switch mode := f.Mode(); {
```

Comment:
if there is an error walking into a directory, the `err` argument will contain that error, so that the walk function (this one) can decide what to do about it. In that case the f argument here will be nil (os.FileInfo is an interface) and this code will likely panic.

Reply:
Hmm - the naive attempt to fix this one doesn't work, as this is not like walking as I expected... and on catching errors here that's obvious (though it didn't panic in this case).

```
 FATAL:   While performing build: packer failed to pack: while unpacking tmpfs: unable to access sandbox path: /tmp/sbuild-621659804/fs/directory
```

Thanks - I'll have to do some more thinking on this, obviously.